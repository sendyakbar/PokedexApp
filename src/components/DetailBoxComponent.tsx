import React from 'react';
import {StyleSheet, View, ViewStyle} from 'react-native';

import InfoRow from '../base/InfoRow';
import Separator from '../base/Separator';

type ItemDataType = {
  label: string;
  value: string;
};

type Props = {
  data: ItemDataType[];
};

export default function DetailBoxComponent(props: Props): JSX.Element {
  const {data} = props;

  return (
    <View style={styles.container}>
      {data.map((d: ItemDataType, i: number) => (
        <>
          <InfoRow label={d.label} value={d.value} key={i} />
          {i !== data.length - 1 ? <Separator /> : null}
        </>
      ))}
    </View>
  );
}

type StyleType = {
  container: ViewStyle;
};

const styles = StyleSheet.create<StyleType>({
  container: {
    backgroundColor: '#eeeeee',
    paddingVertical: 2,
    borderRadius: 4,
  },
});
