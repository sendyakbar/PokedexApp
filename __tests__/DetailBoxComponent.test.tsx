import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import DetailBoxComponent from '../src/components/DetailBoxComponent';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DetailBoxComponent
        data={[
          {label: 'Test', value: 'Success'},
          {label: 'Test2', value: 'Success2'},
        ]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
