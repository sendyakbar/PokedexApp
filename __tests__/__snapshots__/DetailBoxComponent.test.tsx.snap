// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#eeeeee",
      "borderRadius": 4,
      "paddingVertical": 2,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontSize": 14,
            "fontWeight": "700",
          }
        }
      >
        Test
      </Text>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 14,
            "fontWeight": "700",
          }
        }
      >
        :
      </Text>
    </View>
    <View
      style={
        {
          "flex": 2,
          "paddingHorizontal": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "grey",
            "fontSize": 14,
            "fontWeight": "500",
          }
        }
      >
        Success
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#dddddd",
        "borderRadius": 100,
        "height": 0.5,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontSize": 14,
            "fontWeight": "700",
          }
        }
      >
        Test2
      </Text>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 14,
            "fontWeight": "700",
          }
        }
      >
        :
      </Text>
    </View>
    <View
      style={
        {
          "flex": 2,
          "paddingHorizontal": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "grey",
            "fontSize": 14,
            "fontWeight": "500",
          }
        }
      >
        Success2
      </Text>
    </View>
  </View>
</View>
`;
